core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -189,7 +189,7 @@ object DumpLogSegments {
       val wrapperMessageOpt = shallowIter.find(_.offset >= entry.offset + timeIndex.baseOffset)
       if (!wrapperMessageOpt.isDefined || wrapperMessageOpt.get.offset != entry.offset + timeIndex.baseOffset) {
```

Comment:
@hachikuji Why do you want to use match for boolean expressions? 
Since match is efficient but for compactness and performance shouldn't we be using  if/else?